### DISCUSSION TITLE
Internationalization of PyAnsys sphinx document

### DISCUSSION THREAD

**Discussion by tkoyama010** [ansys/pyansys#570] [Ideas]
PyVista has a documentation internationalization project and has created a page in Japanese.

About translating of PyVista: https://docs.pyvista.org/version/stable/#translating
Japanese page: https://pyvista.github.io/pyvista-docs-dev-ja/index.html

This is based on the internationalization feature of Sphinx.
- https://www.sphinx-doc.org/en/master/usage/advanced/intl.html

The documentation of Sphinx itself is also internationalized in this way.
- English page: https://www.sphinx-doc.org/en/master/
- Spanish page: https://www.sphinx-doc.org/es/master/
- Japanese page: https://www.sphinx-doc.org/ja/master/

Could we use PyAnsys documentation to follow suit and internationalize? If people can understand the documentation in their language when they read it, it will be more accessible to them.

I do not have an Ansys license myself so I cannot build the documentation, but I can comment on how it can be internationalized.

**Comment by RobPasMue** [comment]
This might be a good idea we can explore @ansys/pyansys-core - we should definitely give it a try.

**Reply by tkoyama010** [reply]
I noticed that the `make html-noexamples` command allows us to build documentation without a license. I will try translating it first.

**Reply by RobPasMue** [reply]
For which specific package are you trying to build the docs @tkoyama010? =)

**Reply by tkoyama010** [reply]
I am building `pymapdl`. Solid mechanics is my specialty, so I can be useful in translation.

**Reply by RobPasMue** [reply]
That'd be awesome, thanks @tkoyama010!

**Reply by SMoraisAnsys** [reply]
Agreed, really great idea !
@tkoyama010 If you end up with something (even WIP) working, feel free to drop your PR or a link to your changes. That would greatly help us !

**Reply by tkoyama010** [reply]
Yes! I opened the Japanese `pymapdl` page now :tada: :tada: :tada:

Below is the Repository that hosts the Repository and GitHub Pages that summarize the data and procedures for the translation. Once these operations are stable, we would like to ask the Ansys organization to manage them.

1. [Project to provide pymapdl official documentation in multiple languages](https://github.com/tkoyama010/pymapdl-doc-translations)
1. [Repository to hosting Japanese pymapdl documentation GitHub page](https://github.com/tkoyama010/pymapdl-docs-dev-ja)

> [!CAUTION]
> Note that the Japanese page is still being translated. The untranslated sections are in English. Therefore, we are looking for translator members to help with the translation according to the "Contributing to `pymapdl` documentation translation" section of the 1 repository. We will soon have a Spanish page as well, so please actively participate in translating this page as well.

Of course, you can also apply for other languages. Please check the repository description for more information.

[![Screenshot_2024-05-22_02-06-24](https://github.com/ansys/pyansys/assets/7513610/dc25bf3f-c242-4ca6-bf97-080214203557)](https://tkoyama010.github.io/pymapdl-docs-dev-ja/)

**Reply by tkoyama010** [reply]
It says there are 370K words to translate.
![Screenshot_2024-05-22_02-25-56](https://github.com/ansys/pyansys/assets/7513610/941779df-c2ac-480c-acff-714de3ad06c9)

**Reply by RobPasMue** [reply]
Hi @tkoyama010 thanks for demonstrating how this can be done! It's definitely a learning experience for all of us. We will have to see whether we can take over such an effort on our side or not, as it appears that the final translation has to be done "by hand". It my imply a lot of work on our already overloaded dev team. But we will definitely discuss it! :)

**Reply by tkoyama010** [reply]
@RobPasMue 
Thank you for your favorable comments. I understand that the development team is very busy. Rest assured, I plan to maintain the system by myself for at least one to two years to see how the community responds.
1. For now, I will send a Pull Request to `pymapdl` to add to the documentation explaining how to contribute to the translation.
2. It may not be possible right away, but it would be great if we could add a language-switching option to the version number-switching section. I am thinking of posting this as an issue.

**Reply by tkoyama010** [reply]
I started thinking it would be a good idea to merge my po files into pymapdl using the same procedure as pyOpenSci's [python-package-guide/TRANSLATING.md](https://github.com/pyOpenSci/python-package-guide/blob/main/TRANSLATING.md). I am happy to create a Pull Request about it.
